syntax = "proto3";

package namenode;

// Mensajes actualizados.
message SendFileToDataNodeRequest {
  string file_name = 1;
  bytes data = 2;
}

message SendFileToDataNodeResponse {
  bool success = 1;
  string error_message = 2;
}

message ReceiveResponseFromDataNodeRequest {
  string file_name = 1;
  bool success = 2;
}

message ReceiveResponseFromDataNodeResponse {
  bool acknowledged = 1;
  string error_message = 2;
}

message FileRegistryRequest {
}

message FileRegistryResponse {
  repeated FileEntry files = 1;
  bool success = 2;
  string error_message = 3;
}

message FileEntry {
  string file_name = 1;
  repeated string node_addresses = 2;
}

// Servicio NameNode.
service NameNodeService {
  rpc SendFileToDataNode(SendFileToDataNodeRequest) returns (SendFileToDataNodeResponse);
  rpc ReceiveResponseFromDataNode(ReceiveResponseFromDataNodeRequest) returns (ReceiveResponseFromDataNodeResponse);
  rpc GetFileRegistry(FileRegistryRequest) returns (FileRegistryResponse);
}
